Atomic/mount.py

```diff
@@ -732,6 +732,28 @@ def _clean_temp_container_by_path(self, path):
             self.d.remove_container(short_cid)
         self._clean_tmp_image()
 
+
+class DockerMountContextManager(object):
```

Comment:
I *really* love this piece of code.  It could be used quite a bit to replace less efficient code in atomic.  A couple of thoughts:

* Can you write one or more unit tests for it and maybe one integration test?
* Is there any way we could modify this so that if an exception occurred, it would clean up after itself?